api/src/client/dynamics/license-status/DynamicsLicenseHealthCheckClient.ts

```diff
@@ -0,0 +1,54 @@
+import type { AccessTokenClient } from "@client/dynamics/types";
+import type { HealthCheckMetadata, HealthCheckMethod } from "@domain/types";
+import type { LogWriterType } from "@libs/logWriter";
+import axios, { type AxiosError } from "axios";
+
+type Config = {
+  accessTokenClient: AccessTokenClient;
+  orgUrl: string;
+};
+
+export const DynamicsLicenseHealthCheckClient = (
```

Comment:
I'm noticing that each API has a corresponding health check client. Is there a reason we opted for this approach instead of just utilizing the existing resources we built?

Reply:
Dependency inversion, and because each has unique needs that aren't applicable to the others.